main/git/APKBUILD

```diff
@@ -166,8 +166,16 @@ daemon() {
 		"$pkgdir"/$_gitcoredir/git-http-backend \
 		"$pkgdir"/$_gitcoredir/git-shell \
 		"$subpkgdir"/$_gitcoredir/
+}
+
+daemon_openrc() {
+	pkgdesc="Git protocol daemon (OpenRC init scripts)"
+	depends="git-daemon=${pkgver}-r${pkgrel} ${depends_openrc}"
+	install_if="openrc ${pkgname}-daemon=${pkgver}-r${pkgrel}"
+	replaces="git"
 
-	mv "$pkgdir"/etc "$subpkgdir"/
+	mkdir -p "${subpkgdir}/etc"
+	mv "${pkgdir}/etc" "${subpkgdir}/"
```

Comment:
Using default_openrc is a better idea.

Reply:
but doesn't work sadly. Because the openrc package is specifically for git-daemon, abuild doesn't seem to pick up on it. It may be that it only works for the base package? I'm happy to try again if you are convinced this should work with default_openrc. Would an alternative solution work where instead of using _default_openrc I call the external actual function? I may have not tried that combination.

Reply:
The detection for the subpackage is handled by providing the daemon_openrc function. You can call the default_openrc function from within your function to use the logic from abuild instead of moving etc yourself.

Reply:
ah, right; so keep the function, but call default_openrc; that makes sense.